### PR TITLE
S452842_GCN

### PR DESCRIPTION
**Pull request by SteveInUQ**
This is the submission of my GCN implementation. I used TensorFlow to implement a GCN model based on the dataset of Facebook Large Page-Page Network. This project contains following files:

- README.md: Introduction and summary of my project 
- test_drive.py: the main file for training the model
- data_preprocessing.py: contains the class for process the data
- model.py: the architecture of my GCN model
- image foler: contains the visualization of result 

Zhu Gao (s5621842)

### CODE REVIEW

**Comment by SteveInUQ** [comment]
Finish GCN project and push

**Comment by SteveInUQ** [comment]
Update the README file to fix the display error

**Comment by gayanku** [comment]
- Last commit on Nov 4
- Good details in commit messages and PR
- Readme is well done with background, results, diagrams. 
- Good code structure and modularization. Appropriate comments. 
- multi layer (3) layer GCN, Good accuracy (+90%) - _fully supervised_.
**- Valid for NORMAL (subject to using Fully supervised)**

- Used fully supervised split (50%/25%/25%). This may explain the high accuracy. In future make sure that n_train < n_test for Semi supervised.

**Comment by SteveInUQ** [comment]
> * Last commit on Nov 4
> * Good details in commit messages and PR
> * Readme is well done with background, results, diagrams.
> * Good code structure and modularization. Appropriate comments.
> * multi layer (3) layer GCN, Good accuracy (+90%) - _fully supervised_.
>   **- Valid for NORMAL (subject to using Fully supervised)**
> * Used fully supervised split (50%/25%/25%). This may explain the high accuracy. In future make sure that n_train < n_test for Semi supervised.

Thanks for you replying and suggestion

**Comment by SteveInUQ** [comment]
> * Last commit on Nov 4
> * Good details in commit messages and PR
> * Readme is well done with background, results, diagrams.
> * Good code structure and modularization. Appropriate comments.
> * multi layer (3) layer GCN, Good accuracy (+90%) - _fully supervised_.
>   **- Valid for NORMAL (subject to using Fully supervised)**
> * Used fully supervised split (50%/25%/25%). This may explain the high accuracy. In future make sure that n_train < n_test for Semi supervised.

Thanks for your reply and suggestions.

**Comment by SteveInUQ** [comment]
Update the size of the training, validation, test dataset by following the suggestions, and update the visualization of GCN_model loss and t-sne in the README file.

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments, but could have more info -1
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description OK
